test/swagger.js

```diff
@@ -587,3 +587,46 @@ test('swagger json output should not omit enum part in params config', t => {
       })
   })
 })
+
+test('basePath support', t => {
+  t.plan(3)
+  const fastify = Fastify()
+
+  fastify.register(fastifySwagger, {
+    swagger: Object.assign({}, swaggerInfo.swagger, {
+      basePath: '/prefix'
+    })
+  })
+
+  fastify.get('/prefix/endpoint', {}, () => {})
+
+  fastify.ready(err => {
+    t.error(err)
+
+    const swaggerObject = fastify.swagger()
+    t.notOk(swaggerObject.paths['/prefix/endpoint'])
+    t.ok(swaggerObject.paths['/endpoint'])
```

Comment:
Hey Matteo,

because `basePath` is set to `/prefix`, the resulting path should be `/endpoint`. At least that's my interpretation on how it's supposed to be according to the spec: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#paths-object
Also, in the example swagger spec (https://petstore.swagger.io/v2/swagger.json) you'll see that the `paths` don't include the `basePath` prefix of `/v2`...

Patrick